magicblock-config-macro/src/lib.rs

```diff
@@ -158,3 +161,73 @@ pub fn clap_from_serde(_attr: TokenStream, item: TokenStream) -> TokenStream {
     }
     .into()
 }
+
+/// Derives the `Merge` trait for the annotated struct.
+///
+/// **ASSUMES THAT FIELDS WITH NAMES CONTAINING "Config" HAVE A `merge` METHOD**
+///
+/// This macro is used to derive the `Merge` trait for a struct.
+/// The `Merge` trait is used to merge two instances of the struct.
+///
+/// # Example
+/// ```rust
+/// use magicblock_config_macro::Mergeable;
+///
+/// #[derive(Default, Mergeable)]
+/// struct SomeOtherConfig {
+///     field1: u32,
+///     field2: String,
+/// }
+///
+/// #[derive(Default, Mergeable)]
+/// struct MyConfig {
+///     field1: u32,
+///     field2: SomeOtherConfig,
+/// }
+/// ```
+///
+/// Will become:
+/// ```rust
+/// use magicblock_config_helpers::Merge;
+///
+/// #[derive(Default)]
+/// struct SomeOtherConfig {
+///     field1: u32,
+///     field2: String,
+/// }
+///
+/// impl Merge for SomeOtherConfig {
+///     fn merge(&mut self, other: Self) {
+///         let default = Self::default();
+///         if self.field1 == default.field1 && other.field1 != default.field1 {
```

Comment:
Make sure to update these docs again after removing the superfluous second condition.

Also BTW some tests that actually check the code that is rendered using `quote` (kind of the opposite of what I'm doing [in shank](https://github.com/metaplex-foundation/shank/blob/master/shank-macro-impl/src/parsed_enum/parsed_enum.rs#L143) would make this a living documentation that will fail once we update the implementation.